### PR TITLE
mpl: pass MPL's CFLAGS back to MPICH

### PR DESCRIPTION
**Pull request by yfguo**
## Pull Request Description

PR#7074 consolidated SSE2 and AVX related optimization options into MPL's configure because only MPL explicitly use them. MPICH's CFLAGS no longer has those options. However, with Intel compilers, this does results in some performance degradation. Therefore, we need to propagate the CFLAGS in MPL back into the MPICH configure.

This PR creates a `MPL_CFLAGS` in the localdefs that allows the PAC_CONFIG_MPL macro to add it to the main CFLAGS.

This is an alternative and probably better solution than the 2nd commit in #7150 .

## Author Checklist
* [ ] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [ ] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [ ] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by yfguo** [comment]
test:mpich/ch4/ofi
test:mpich/ch4/ucx

**Comment by yfguo** [comment]
test:mpich/ch4/ofi
test:mpich/ch4/ucx

File: src/mpl/configure.ac [hunk 1]
```diff
@@ -1405,6 +1406,9 @@ AX_GCC_VAR_ATTRIBUTE(used)
 
 PAC_POP_ALL_FLAGS
 
+MPL_CFLAGS="${CFLAGS#${PARENT_CFLAGS}}"
```
Comment (hzhou) [inline src/mpl/configure.ac:1409]: Could you add a short comment explaining what the substitution does?

**Review by hzhou — APPROVED** [review]
Looks good. Just two minor comments.

File: src/mpl/configure.ac (hunk 1 above)
Comment (yfguo) [reply src/mpl/configure.ac:1409]: This is supported in POSIX. See https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html, section 2.6.2 Parameter Expansion
```${parameter#[word]}
Remove Smallest Prefix Pattern. The word shall be expanded to produce a pattern. The parameter expansion shall then result in parameter, with the smallest portion of the prefix matched by the pattern deleted. If present, word shall not begin with an unquoted '#'.```

**Pull request merged**